core/src/main/scala/cats/instances/invariant.scala

```diff
@@ -19,9 +18,7 @@ trait InvariantMonoidalInstances {
     def pure[A](a: A): Semigroup[A] = new Semigroup[A] {
       def combine(x: A, y: A): A = a
       override def combineAllOption(as: TraversableOnce[A]): Option[A] =
-        if (as.isEmpty) None
-        else if (as.size == 1) as.toList.headOption
-        else Some(a)
+        if (as.size > 1) Some(a) else as.toTraversable.headOption
```

Comment:
There may be merit to these changes, but they seem to be unrelated to the rest of the PR, and I think that they should be separated out into another PR. I'm not sure if there are types that implement `TraversableOnce` for which an `isEmpty` check would be significantly faster than a `size` check on an empty collection, but it's something that people might not think about when reviewing this PR.

Reply:
Np, I will revert these changes, they are indeed not related to `Semigroup.instance`

Reply:
@kailuowang that's true, but unfortunately not all of us are in Java 8/Scala 2.12 lands yet

@ceedubs 
I can write jmh benchmarks for both cases - the old way of creating instances  and the new one using `Semigroup.instance` 
 OR
 alternatively I can revert those changes and just leave  addition of `Semigroup.instance` , as this is the main goal here. Then we can try  another PR with benchmarks or just let it go
Please let me know

Reply:
> I can revert those changes and just leave addition of Semigroup.instance , as this is the main goal here. Then we can try another PR with benchmarks or just let it go

@jozic this sounds like the path of least resistance to me. Just please be sure to add a test for the `Semigroup.instance` method, since I think that currently it's only indirectly tested through the `Semigroup` instances that you are using it to create.